apps/src/accounts/AccountInformation/index.tsx

```diff
@@ -0,0 +1,393 @@
+import classNames from 'classnames';
+import React, {useMemo, useState} from 'react';
+
+import TextField from '@cdo/apps/componentLibrary/textField/TextField';
+import i18n from '@cdo/locale';
+
+import {hashEmail} from '../../code-studio/hashEmail';
+import Alert, {alertTypes} from '../../componentLibrary/alert/Alert';
+import {Button} from '../../componentLibrary/button';
+import {SimpleDropdown} from '../../componentLibrary/dropdown';
+import Link from '../../componentLibrary/link/Link';
+import {Heading2} from '../../componentLibrary/typography';
+import {getAuthenticityToken} from '../../util/AuthenticityTokenStore';
+import ChangeEmailModal from '../ChangeEmailModal';
+
+import {AccountInformationProps} from './types';
+
+import styles from './style.module.scss';
+
+export const AccountInformation: React.FC<AccountInformationProps> = ({
+  authorizedTeacher,
+  secretPictureAccountOnly,
+  teacherManagedAccount,
+  parentManagedAccount,
+  shouldSeeEditEmailLink,
+  isPasswordRequired,
+  isStudent,
+  migrated,
+  userType,
+  userAge,
+  userUsername,
+  userDisplayName,
+  userProperties,
+  userEmail,
+  hashedEmail,
+  encryptedPasswordPresent,
+  canEditPassword,
+  sponsored,
+  ageDropdownOptions,
+  studentInLockoutFlow,
+  showGenderInput,
+  usStateDropdownOptions,
+  countryCode,
+  isUSA,
+}) => {
+  const [name, setName] = useState(userDisplayName ?? '');
+  const [username, setUsername] = useState(userUsername ?? '');
+  const [email, setEmail] = useState(userEmail ?? '');
+  const [gender, setGender] = useState(
+    userProperties?.gender_student_input ?? ''
+  );
+  const [age, setAge] = useState(userAge ?? '');
+  const [usState, setUsState] = useState(userProperties?.us_state ?? '');
+  const [password, setPassword] = useState('');
+  const [passwordConfirmation, setPasswordConfirmation] = useState('');
+  const [currentPassword, setCurrentPassword] = useState('');
+  const [showChangeEmailModal, setShowChangeEmailModal] = useState(false);
+  const [errors, setErrors] = useState<Record<string, string[]>>({});
+
+  const displayNameHelperMessage = useMemo(
+    () =>
+      isStudent
+        ? undefined
+        : i18n.accountInformation_display_name_hint_teacher(),
+    [isStudent]
+  );
+
+  const emailValue = useMemo(
+    () => (isStudent ? `***${i18n.encrypted()}***` : email ?? ''),
+    [isStudent, email]
+  );
+
+  const usStateOptions = useMemo(
+    () =>
+      [[i18n.selectState(), '']]
+        .concat(usStateDropdownOptions)
+        .map(([text, value]) => ({text, value})),
+    [usStateDropdownOptions]
+  );
+
+  const handleSubmitAccountSettingsUpdate = async () => {
+    const userUpdates = {
+      name,
+      username,
+      password,
+      password_confirmation: passwordConfirmation,
+      current_password: currentPassword,
+      age: isStudent ? age : '21+',
+      gender_student_input: isStudent && gender ? gender : undefined,
```

Comment:
Nit: Could this theoretically be `gender ?? undefined` since a teacher would not have the field to have set their gender in the first place?

Reply:
I think this is a good point! Although I like that this is explicit, in that you don't need to have inherent knowledge of which user types store which values to know what this code does. I also wonder about when a student changes to a teacher account, or vise versa? probably not an issue either way, but this is more clearly tied to the user type

Reply:
Sounds good!